File: C#/2171. Removing Minimum Number of Magic Beans.cs
public class Solution {
    public long MinimumRemoval(int[] beans) {
        long n = beans.Length;
        long res = long.MaxValue;
        long sum = 0;
        Array.Sort(beans);
        foreach(int i in beans){
            sum += i;
        }
        for(int i = 0; i<n; i++){
            res = Math.Min(res, sum - (n - i) * beans[i]);
        }
        return res;
    }
}

File: Java/2171. Removing Minimum Number of Magic Beans.java
class Solution {
    public long minimumRemoval(int[] beans) {
        long n = beans.length;
        long res = Long.MAX_VALUE;
        long sum = 0;
        Arrays.sort(beans);
        for(int i : beans){
            sum += i;
        }
        for(int i = 0; i<n; i++){
            res = Math.min(res, sum - (n - i) * beans[i]);
        }
        return res;
    }
}
